packages/plugin-core/src/_extension.ts

```diff
@@ -1255,6 +1258,54 @@ function _setupLanguageFeatures(context: vscode.ExtensionContext) {
 }
 
 // ^qxkkg70u6w0z
+
+function _setupTreeViewCommands(
+  treeView: NativeTreeView,
+  existingCommands: string[]
+) {
+  if (
+    !existingCommands.includes(DENDRON_COMMANDS.TREEVIEW_LABEL_BY_TITLE.key)
+  ) {
+    vscode.commands.registerCommand(
+      DENDRON_COMMANDS.TREEVIEW_LABEL_BY_TITLE.key,
+      sentryReportingCallback(() => {
+        treeView.updateLabelType({
+          labelType: TreeViewItemLabelTypeEnum.title,
+        });
+      })
+    );
+  }
+
+  if (
+    !existingCommands.includes(DENDRON_COMMANDS.TREEVIEW_LABEL_BY_FILENAME.key)
+  ) {
+    vscode.commands.registerCommand(
+      DENDRON_COMMANDS.TREEVIEW_LABEL_BY_FILENAME.key,
+      sentryReportingCallback(() => {
+        treeView.updateLabelType({
+          labelType: TreeViewItemLabelTypeEnum.filename,
+        });
+      })
+    );
+  }
+
+  /**
+   * This is a little flaky right now, but it works most of the time.
+   * Leaving this for dev / debug purposes.
+   * Enablement is set to be DendronContext.DEV_MODE
+   *
+   * TODO: fix tree item register issue and flip the dev mode flag.
```

Comment:
out of curiosity, the flakiness is that sometimes it throws that 'duplicate id registration' error?

Reply:
That is _mostly_ fixed. it still happens but very rarely it seems. I removed the cause for like 95% of the cases but I'm yet to figure out the rest.

Also not mentioned here, but we are at worst calling n (numNotes) different async callbacks to fetch the file so it's pretty slow. The main reason VSCode doesn't provide this as a standard API.

I would ideally want to change this to be tree item specific (possible, add it as an inline button for each item), or have it only expand visible tree items (not sure if possible). Haven't specced out the journey here yet.

I still would like to have this behind the dev flag even if we make it tree item specific though, as this is pretty useful for visually debugging the tree view.

Reply:
cool, thanks for the detailed explanation.